update_secrets.py

```diff
@@ -0,0 +1,135 @@
+import copy
+import inspect
+import os
+import subprocess
+import optparse
+import sys
+from time import sleep
+import dotenv
+import yaml
+import re
+
+class Updater(object):
+
+    def __init__(self, configfile='.env', envfile='kubernetes-envs.yml'):
+        '''Initialize.
+        @param: configfile a .env style config file. See README for more.
+        '''
+        if os.path.exists(configfile):
+            # we set ourselves as load_dotenv has system env variables to take
+            # precedence which in our experience is confusing as a user changes a
+            # var and re-runs and nothing happens
+            # dotenv.load_dotenv('.env')
+            out = dotenv.main.dotenv_values(configfile)
+            # we need stuff in the environment for docker
+            os.environ.update(out)
+        self.envs = os.environ
+        self.configs = yaml.load(open(envfile).read())
+        self.configs = self._update_with_envs(self.configs, self.envs)
+
+    def _update_with_envs(self, configfile, env):
+        for service in configfile:
+            for env_ in configfile[service]['environment']:
+                if not configfile[service]['environment'].get(env_):
+                    configfile[service]['environment'][env_] = env.get(env_, '')
+                else:
+                    pattern = re.compile(r'\${.*}')
+                    env_value = configfile[service]['environment'][env_]
+                    to_replace = pattern.findall(str(env_value))
+                    for rpl in to_replace:
+                        configfile[service]['environment'][env_] = env.get(rpl[2:-1], '')
+        return configfile
+
+
+    def _run_commands(self, cmd, options=''):
+        out = ''
+        cmd = cmd.split(' ') + [options]
```

Comment:
just use split():
```
>>> "sfdsfds     fsdfsdds   fds".split()
['sfdsfds', 'fsdfsdds', 'fds']
>>> "sfdsfds     fsdfsdds   fds".split(" ")
['sfdsfds', '', '', '', '', 'fsdfsdds', '', '', 'fds']
```

Reply:
And [`shlex.split`](https://docs.python.org/3/library/shlex.html#shlex.split) is even better.